sklearn/metrics/_pairwise_distances_reduction/_radius_neighborhood.pyx.tp

```diff
@@ -232,9 +232,6 @@ cdef class RadiusNeighbors{{name_suffix}}(BaseDistanceReducer{{name_suffix}}):
         ITYPE_t X_start,
         ITYPE_t X_end,
     ) nogil:
-        cdef:
-            ITYPE_t idx, jdx
```

Comment:
`idx` is used in a few line `for idx in range(X_start, X_end):`. Is Cython inferring the type of `idx`?

Reply:
Good catch, that's a mistake from me!

`idx`'s type is indeed inferred as a C `int` (see `__pyx_t_1` in the collapsed item).

<details>
<summary>Generated C++ code</summary>

```c++

static void __pyx_f_7sklearn_7metrics_29_pairwise_distances_reduction_20_radius_neighborhood_17RadiusNeighbors64__parallel_on_X_prange_iter_finalize(struct __pyx_obj_7sklearn_7metrics_29_pairwise_distances_reduction_20_radius_neighborhood_RadiusNeighbors64 *__pyx_v_self, CYTHON_UNUSED __pyx_t_7sklearn_5utils_9_typedefs_ITYPE_t __pyx_v_thread_num, __pyx_t_7sklearn_5utils_9_typedefs_ITYPE_t __pyx_v_X_start, __pyx_t_7sklearn_5utils_9_typedefs_ITYPE_t __pyx_v_X_end) {
  __pyx_t_7sklearn_5utils_9_typedefs_ITYPE_t __pyx_v_idx;
  int __pyx_t_1;
  __pyx_t_7sklearn_5utils_9_typedefs_ITYPE_t __pyx_t_2;
  __pyx_t_7sklearn_5utils_9_typedefs_ITYPE_t __pyx_t_3;
  __pyx_t_7sklearn_5utils_9_typedefs_ITYPE_t __pyx_t_4;

  /* "sklearn/metrics/_pairwise_distances_reduction/_radius_neighborhood.pyx":220
 *     ) nogil:
 *         # Sorting neighbors for each query vector of X
 *         if self.sort_results:             # <<<<<<<<<<<<<<
 *             for idx in range(X_start, X_end):
 *                 simultaneous_sort(
 */
  __pyx_t_1 = (__pyx_v_self->sort_results != 0);
  if (__pyx_t_1) {

    /* "sklearn/metrics/_pairwise_distances_reduction/_radius_neighborhood.pyx":221
 *         # Sorting neighbors for each query vector of X
 *         if self.sort_results:
 *             for idx in range(X_start, X_end):             # <<<<<<<<<<<<<<
 *                 simultaneous_sort(
 *                     deref(self.neigh_distances)[idx].data(),
 */
    __pyx_t_2 = __pyx_v_X_end;
    __pyx_t_3 = __pyx_t_2;
    for (__pyx_t_4 = __pyx_v_X_start; __pyx_t_4 < __pyx_t_3; __pyx_t_4+=1) {
      __pyx_v_idx = __pyx_t_4;
```

</details>

I think that's possible for Cython since `X_start`, `X_end` are integer but that's auto-magic.

I would revert this change to explicitly type it like was done before:

```suggestion
        cdef:
            ITYPE_t idx

```